pkg/frontend/validate.go

```diff
@@ -44,3 +44,20 @@ func (f *frontend) validateSubscriptionState(ctx context.Context, key string, al
 
 	return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionState, "", "Request is not allowed in subscription in state '%s'.", doc.Subscription.State)
 }
+
+// validateOpenShiftUniqueKey returns which unique key if causing a 412 error
+func (f *frontend) validateOpenShiftUniqueKey(ctx context.Context, doc *api.OpenShiftClusterDocument) error {
```

Comment:
Can we have unit test for this, please? :)

Reply:
added